setup.py

```diff
@@ -34,7 +34,7 @@
 
 WITH_DISTRIBUTED = not check_env_flag('NO_DISTRIBUTED') and not IS_WINDOWS
 WITH_DISTRIBUTED_MW = WITH_DISTRIBUTED and check_env_flag('WITH_DISTRIBUTED_MW')
-WITH_GLOO_IBVERBS = check_env_flag('WITH_GLOO_IBVERBS')
+WITH_GLOO_IBVERBS = WITH_DISTRIBUTED and not check_env_flag('NO_GLOO_IBVERBS')
```

Comment:
isn't it the case that you need some ibverbs headers to build it with IB support and they are not generally installed on systems that don't have them? Let's just do the test as a best-effort way of detecting infiniband locally and fall back to false otherwise.